wiki/wiki/doctype/wiki_page/wiki_page.py

```diff
@@ -216,6 +216,26 @@ def calculate_toc_html(self, html):
 	def get_context(self, context):
 		self.verify_permission("read")
 		self.set_breadcrumbs(context)
+		results = frappe.db.get_all("User Permission", {"user": frappe.session.user, "allow": "Wiki Space"}, ["user", "allow", "for_value", "name"])
+		if results:
+			match_found = False
+			for result in results:
+				wikiGroup = frappe.db.get_all("Wiki Group Item", {"parent": result['for_value']},["name","wiki_page"])
+				if wikiGroup:
+					for wk in wikiGroup:
+						if wk['wiki_page'] == self.name:
+							match_found = True
+							break
```

Comment:
This is too much nesting. Please fix it. You can refer https://gist.github.com/18alantom/d9f0565c0f42d6a71311d4a3093a1331 to refactor this.